### PR TITLE
Make `parse-backticks` more reliable

### PR DESCRIPTION
**Pull request by kidonng**
On request by https://github.com/sindresorhus/refined-github/pull/3290#issuecomment-659262450

Part of #3084

~~Edited the title and it stil runs well!!! 😋 (though there's about 2 seconds' delay)~~ (Never mind, that is `format-conversation-titles`)

### CODE REVIEW

File: source/features/parse-backticks.tsx [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 import features from '.';
-import observeElement from '../helpers/simplified-element-observer';
 import {parseBackticks} from '../github-helpers/dom-formatters';
 
 function parse(selectors: string[]): void {
-	for (const element of select.all(selectors.map(selector => selector + ':not(.rgh-backticks-already-parsed)'))) {
-		element.classList.add('rgh-backticks-already-parsed');
-		parseBackticks(element);
+	for (const selector of selectors) {
+		observe(selector, {
```
Comment (fregante) [inline source/features/parse-backticks.tsx:10]:
```suggestion
	observe(selectors.join(','), {
```

I'm pretty sure `rgh-backticks-already-parsed` is still needed; last time I tried that was exactly the issue. Try for example:
1. load this PR
2. visit another PR tab (to trigger AJAX navigation)
3. click the browser's back button

You might see that the feature run twice on the same piece of DOM

Comment (fregante) [reply source/features/parse-backticks.tsx:10]: Also you can merge all the `init`s into one so there's a single `observe` for the whole feature

Comment (kidonng) [reply source/features/parse-backticks.tsx:10]: > 3. click the browser's back button

Usually there need to be a 4th step to trigger that: then click the forward button. But whatever.

**Comment by kidonng** [comment]
We have to test the selectors and narrow scope down since they are now global

**Comment by kidonng** [comment]
- Grouped/merged related selectors
- Replace some generic selectors with more specific selectors
- Add detailed comments

File: source/features/parse-backticks.tsx [hunk 2]
```diff
@@ -1,113 +1,54 @@
...
-	]);
-}
-
-function initUserProfile(): void {
-	parse([
+		'.commits blockquote', // `isDashboard` newsfeed commits
+		'.notifications-list-item p.text-normal', // `isNotifications` issue and PR title
 		'.profile-timeline-card .text-gray-dark', // `isUserProfileMainTab` issue and PR title
-		'[itemprop="description"]' // `isUserProfileRepoTab` repository description
+		'[itemprop="description"]', // `isUserProfileRepoTab` repository description
```
Comment (fregante) [inline source/features/parse-backticks.tsx:42]: Indeed this is probably too broad now

**Comment by yakov116** [comment]
Not sure if you fixed this but 

image

**Comment by kidonng** [comment]
Yeah it's parsed

image

Isn't the selector for that place is already there before this PR?

**Comment by yakov116** [comment]
Funny it was not working for me before

File: source/features/parse-backticks.tsx [hunk 3]
```diff
@@ -1,113 +1,54 @@
...
+		add(element) {
+			element.classList.add('rgh-backticks-already-parsed');
+			parseBackticks(element);
+		}
+	});
 }
 
-function initRepo(): void {
+function init(): void {
 	parse([
```
Comment (fregante) [inline source/features/parse-backticks.tsx:17]: You can avoid the extra `parse` function as well, just merge it into `init`

**Comment by yakov116** [comment]
Should we make a line break in between each unique page type? (Ex: `isWiki` `isDashboard`) I think it will make it a drop easier to read.

**Comment by fregante** [comment]
I don't think there's any way to make it any more readable. It's a list of wildly different selectors, it's just what it is

**Comment by fregante** [comment]
Looks great! Possible additions:

- use `selector-observer` in https://github.com/sindresorhus/refined-github/blob/master/source/features/linkify-code.tsx (this would fix https://github.com/sindresorhus/refined-github/issues/2091)
- remove parse backticks from https://github.com/sindresorhus/refined-github/blob/master/source/features/format-conversation-titles.tsx
- either rename `format-conversation-titles` back to `linkify-titles` or merge it into `linkify-code`

**Comment by yakov116** [comment]
This is awesome!! https://github.com/tanmayrajani/notifications-preview-github is now parsed too!!

**Comment by kidonng** [comment]
Is it https://github.com/sindresorhus/refined-github/pull/3397/files#diff-7b0a2c8650978a85cebddd08a46e40baR31? I don't have the extension

**Comment by yakov116** [comment]
Yes.

You should get it, its an awesome extension.

**Pull request merged**